SDL/MacOSBarItems.mm

```diff
@@ -0,0 +1,326 @@
+//
+//  MacOSBarItems.mm
+//  PPSSPP
+//
+//  Created by Serena on 06/02/2023.
+//
+
+#include "Common/System/System.h"
+#include "Common/System/NativeApp.h"
+#include "Core/Config.h"
+#include "Common/Data/Text/I18n.h"
+#include "Common/StringUtils.h"
+#import <AppKit/AppKit.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// NSMenuItem requires the use of an objective-c selector (aka the devil's greatest trick)
+// So we have to make this class
+@interface BarItemsManager : NSObject
++(instancetype)sharedInstance;
+-(void)setupAppBarItems;
+@property (assign) NSMenu *openMenu;
+@property (assign) std::shared_ptr<I18NCategory> mainSettingsLocalization;
+@end
+
+void initBarItemsForApp() {
+    [[BarItemsManager sharedInstance] setupAppBarItems];
+}
+
+// im soooooo sorry for whoever had to read this impl
+@implementation BarItemsManager
++ (instancetype)sharedInstance {
+    static BarItemsManager *stub;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        stub = [BarItemsManager new];
+        stub.mainSettingsLocalization = GetI18NCategory("MainSettings");
+    });
+    
+    return stub;
+}
+
+-(void)setupAppBarItems {
+    NSMenuItem *openMenuItem = [[NSMenuItem alloc] init];
+    openMenuItem.submenu = [self makeOpenSubmenu];
+    
+    NSMenuItem *graphicsMenuItem = [[NSMenuItem alloc] init];
+    graphicsMenuItem.submenu = [self makeGraphicsMenu];
+    
+    NSMenuItem *audioMenuItem = [[NSMenuItem alloc] init];
+    audioMenuItem.submenu = [self makeAudioMenu];
+    
+    [NSApplication.sharedApplication.menu addItem:openMenuItem];
+    [NSApplication.sharedApplication.menu addItem:graphicsMenuItem];
+    [NSApplication.sharedApplication.menu addItem:audioMenuItem];
+}
+
+-(NSMenu *)makeOpenSubmenu {
+    NSMenu *menu = [[NSMenu alloc] initWithTitle:@"File"];
```

Comment:
Some of these have translation strings in the language ini.  I wouldn't try to deal with it for this pull, but for a next step it'd be nice to have the language of the menu magically change when you change languages within PPSSPP.  This happens on Windows.

Although, you might have to strip `&` out of strings from the translation, because it uses those (Windows convention) to indicate the shortcut key for the menu.

-[Unknown]